numpy/lib/format.py

```diff
@@ -682,6 +687,15 @@ def read_array(fp, allow_pickle=True, pickle_kwargs=None):
 
     # Now read the actual data.
     if dtype.hasobject:
+        if allow_pickle is None:
+            # NumPy 1.17.0: remove in 1.19 after deprecation
+            allow_pickle = True
+            msg = ("Object arrays will not be saved by default in the future"
+                    " because `allow_pickle` will default to False. You should"
+                    " add `allow_pickle=True` explicitly to eliminate this"
+                    " warning.")
+            warnings.warn(msg, FutureWarning, stacklevel=2)
```

Comment:
Is this warning in the right place? This warning is about *saving*, but it's inside a function called `read_array`. I think you may have mixed up reading/writing.

Reply:
Nice catch. Changing to default=False, which implies no warning, only an error.

Reply:
the warning belongs on `write_array` instead